animation_nodes/nodes/mesh/mesh_points_scatter.py

```diff
@@ -4,17 +4,26 @@
 from ... utils.math import cantorPair
 from ... events import propertyChanged
 from ... base_types import AnimationNode
-from ... data_structures import Vector3DList, VirtualDoubleList
-from ... algorithms.mesh.points_scatter import randomPointsScatter
+from ... data_structures import Matrix4x4List, VirtualDoubleList
+from ... data_structures.meshes.mesh_data import calculatePolygonNormals
+from ... algorithms.mesh.points_scatter import randomPointsScatter, randomPointsScatterForEdges
 from ... algorithms.mesh.triangulate_mesh import (
     triangulatePolygonsUsingFanSpanMethod,
     triangulatePolygonsUsingEarClipMethod
 )
 
+modeItems = [
+    ("EDGES", "Edges", "Scatter points on edges", "NONE", 0),
+    ("POLYGONS", "Polygons", "Scatter points on polygons", "NONE", 1)
+]
+
 class MeshPointsScatterNode(bpy.types.Node, AnimationNode):
     bl_idname = "an_MeshPointsScatterNode"
     bl_label = "Mesh Points Scatter"
 
+    mode: EnumProperty(name = "Mode", default = "POLYGONS",
+        items = modeItems, update = AnimationNode.refresh)
+
     methodType: BoolProperty(name = "Use Advanced Method for Mesh sampling", default = False,
```

Comment:
While at it, rename this to `useAdvancedTriangulationMethod` and mention triangulation in the name.

Reply:
Is it correct `useAdvancedTriangulationMethod: BoolProperty(name = "Use Ear Clip Triangulation Method",
                                    default = False, update = propertyChanged)` ?

Reply:
Yes, looks good.

Reply:
Okay.